### ISSUE TITLE
Intermittent UI freeze while using Visual Studio Code on WSL2

### ISSUE THREAD

**Issue by joel-rieke** [microsoft/WSL#8191]
### Version

5.10.102.1-microsoft-standard-WSL2+

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.102.1-microsoft-standard-WSL2+

### Distro Version

Debian 11

### Other Software

Visual Studio Code 1.65.2

### Repro Steps

Just use VS Code normally -- debugging, etc...  Also have Google Chrome running along with few other apps.  Sometimes when resizing VSCode, it triggers a freeze, sometimes just trying to drag the window...

Once VSCode freezes, the UI elements become frozen in place and cannot be destroyed even after killing the process.

The only way to recover is to wsl --shutdown

### Expected Behavior

No screen freezing.  But even if it does, I should be able to pkill or kill -9 the app and have the gui elements removed.

### Actual Behavior

GUI elements frozen to the UI and cannot be moved or gotten rid of.

### Diagnostic Logs

I've had this problem for many WSL releases...  I happened to download the WSL kernel and build a custom build.  I had enabled the WSL Debug Console and saw the following leading up to and soon after the crash.  Not sure if this helps much, but it looked interesting...

[   15.217207] dxgk:err: is_handle_valid Entry is freed 40000080 2                                                                                                                                   [   15.217208] dxgk:err: hmgrtable_get_object_by_type invalid handle 40000080                                                                                                                        [   15.221842] dxgk:err: adapter_by_handle failed 40000080                                                                                                                                           
[   49.717162] hv_balloon: Max. dynamic memory size: 5120 MB

[18130.574994] potentially unexpected fatal signal 6.
[18130.576988] CPU: 6 PID: 2151 Comm: Xwayland Not tainted 5.10.102.1-microsoft-standard-WSL2+ #1
[18130.580094] RIP: 0033:0x7fc6679e1115
[18130.581386] Code: c2 b8 ea 00 00 00 0f 05 48 3d 00 f0 ff ff 77 3d 41 89 c0 41 ba 08 00 00 00 31 d2 4c 89 ce bf 02 00 00 00 b8 0e 00 00 00 0f 05 <48> 8b 84 24 08 01 00 00 64 48 33 04 25 28 00 00 00 75 24 44 89 c0
[18130.588063] RSP: 002b:00007ffc748be4c0 EFLAGS: 00000246 ORIG_RAX: 000000000000000e
[18130.590804] RAX: 0000000000000000 RBX: 000056409c311960 RCX: 00007fc6679e1115
[18130.593349] RDX: 0000000000000000 RSI: 00007ffc748be4c0 RDI: 0000000000000002
[18130.595908] RBP: 000056409b05d45e R08: 0000000000000000 R09: 00007ffc748be4c0
[18130.598496] R10: 0000000000000008 R11: 0000000000000246 R12: 00007ffc748be750
[18130.601110] R13: 000056409c8630b0 R14: 00007ffc748bea4c R15: 000056409c8630a0
[18130.603697] FS:  00007fc667372440 GS:  0000000000000000

**Comment by joel-rieke** [comment]
One interesting thing.  After restarting wsl with the custom build, I noticed that wsl.exe -status was incorrectly showing the old version WSL, which was different from "cat /proc/version".  After the crash the two seemed to align and agree on the version of WSL running.  WIll see if things have stabilized since and if they have, I will close this.

**Comment by joel-rieke** [comment]
Nope, still seeing intermittent freezes.  Here's another dump from WSL debug console:

I'm not sure how helpful these dumps are in troubleshooting, but hopeful that they could maybe help.

[ 3570.240006] Code: c2 b8 ea 00 00 00 0f 05 48 3d 00 f0 ff ff 77 3d 41 89 c0 41 ba 08 00 00 00 31 d2 4c 89 ce bf 02 00 00 00 b8 0e 00 00 00 0f 05 <48> 8b 84 24 08 01 00 00 64 48 33 04 25 28 00 00 00 75 24 44 89 c0
[ 3570.246596] RSP: 002b:00007ffdfea16cf0 EFLAGS: 00000246 ORIG_RAX: 000000000000000e
[ 3570.249314] RAX: 0000000000000000 RBX: 0000558e9228e960 RCX: 00007f7f58766115
[ 3570.251802] RDX: 0000000000000000 RSI: 00007ffdfea16cf0 RDI: 0000000000000002
[ 3570.254285] RBP: 0000558e90ad645e R08: 0000000000000000 R09: 00007ffdfea16cf0
[ 3570.256884] R10: 0000000000000008 R11: 0000000000000246 R12: 00007ffdfea16f80
[ 3570.259488] R13: 0000558e927be030 R14: 00007ffdfea1727c R15: 0000558e927be020
[ 3570.262064] FS:  00007f7f580f7440 GS:  0000000000000000
[ 3571.846971] potentially unexpected fatal signal 11.
[ 3571.849529] CPU: 8 PID: 7292 Comm: code Not tainted 5.10.102.1-microsoft-standard-WSL2+ #1
[ 3571.852340] RIP: 0033:0x7faa0c31e969
[ 3571.854079] Code: f6 eb 23 4c 89 f0 4c 09 e0 48 c1 e8 20 74 0d 4c 89 f0 31 d2 49 f7 f4 48 89 d6 eb 0a 44 89 f0 31 d2 41 f7 f4 89 d6 49 8b 45 00 <48> 8b 04 f0 48 85 c0 0f 84 99 01 00 00 44 0f b6 2b 41 f6 c5 01 74
[ 3571.862672] RSP: 002b:00007fffaf989bc0 EFLAGS: 00010202
[ 3571.864517] RAX: 0000000000000000 RBX: 000026420028d408 RCX: 0000000000000001
[ 3571.867034] RDX: 0000000000000001 RSI: 0000000000000001 RDI: e85e064357d7d32e
[ 3571.869628] RBP: 00007fffaf989c00 R08: c3a5c85c97cb3127 R09: 742f627568746947
[ 3571.872215] R10: f29b090fbfa5a2eb R11: dcc5d7ad46a5a654 R12: 0000000000000002
[ 3571.874764] R13: 00007faa0c359580 R14: b0175878a5f18d25 R15: 000026420028d409
[ 3571.877396] FS:  00007faa11dd7080 GS:  0000000000000000

**Comment by joel-rieke** [comment]
This continues to be a problem.  I always apply the latest windows updates.  But alas, still seeing freezes.

**Comment by owenschupp** [comment]
Going to transfer this issue to wslg based off of seeing Xwayland crash in the logs.
